br/pkg/lightning/backend/local/engine.go

```diff
@@ -1034,7 +1061,33 @@ func (w *Writer) appendRowsSorted(kvs []common.KvPair) error {
 		}
 		kvs = newKvs
 	}
-	return w.writer.writeKVs(kvs)
+	tempKvs := make([]common.KvPair, 0)
+	tempKvs = append(tempKvs, kvs[0])
```

Comment:
It's unnecessary to use a `tempKvs` here. We record the start kv index and make slice of the original kvs. 

`appendRows` is on the critical path, so it's worth avoiding unnecessary allocations.